plugin/trino-password-authenticators/src/test/java/io/trino/plugin/password/ldap/TestLdapAuthenticator.java

```diff
@@ -13,38 +13,56 @@
  */
 package io.trino.plugin.password.ldap;
 
-import com.google.common.collect.HashMultimap;
-import com.google.common.collect.ImmutableSet;
-import io.trino.plugin.password.Credential;
+import com.google.common.io.Closer;
 import io.trino.spi.security.AccessDeniedException;
 import io.trino.spi.security.BasicPrincipal;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
-import javax.naming.NamingException;
-
-import java.util.HashSet;
-import java.util.Optional;
-import java.util.Set;
-
+import static io.trino.plugin.password.ldap.LdapUtil.randomSuffix;
+import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 import static org.testng.Assert.assertEquals;
 
 public class TestLdapAuthenticator
 {
-    private static final String BASE_DN = "base-dn";
-    private static final String PATTERN_PREFIX = "pattern::";
+    private final Closer closer = Closer.create();
+
+    private TestingOpenLdapServer openLdapServer;
+    private LdapAuthenticatorClient client;
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        openLdapServer = new TestingOpenLdapServer();
+        closer.register(openLdapServer);
+        openLdapServer.start();
+
+        client = new JdkLdapAuthenticatorClient(new LdapConfig()
+                .setLdapUrl(openLdapServer.getLdapUrl()));
+    }
+
+    @AfterClass
```

Comment:
`@AfterClass(alwaysRun = true)` We should have static analysis for this.

Reply:
It'd be interesting to know why `true` is not the default

Reply:
Maybe to reduce the burden on testng if the tests were failed or skipped ?